pkg/repository/maintenance.go

```diff
@@ -116,8 +116,14 @@ func GetMaintenanceResultFromJob(cli client.Client, job *batchv1.Job) (string, e
 		return "", fmt.Errorf("no pod found for job %s", job.Name)
 	}
 
-	// we only have one maintenance pod for the job
-	return podList.Items[0].Status.ContainerStatuses[0].State.Terminated.Message, nil
+	pod := podList.Items[0]
+	statuses := pod.Status.ContainerStatuses
+	if len(statuses) == 0 {
+		return "", fmt.Errorf("no container statuses found for pod %s", pod.Name)
```

Comment:
Since function wants result from job, what about changing error message to something like
```
container statuses not yet populated for <job>
```
?